File: 백준 Python/Step7 함수/pro7_2.py
#답은 맞는 거 같은데 백준 채점 시간초과 뜸.. 내일 다시!!!!

num = list(range(10000))

for n in range(10000) :
    while True :    #n이 초기생성자로 d(n) 무한 반복
        gen = n + n%10 + n//10 + n//100 + n//1000 + n//10000
        #print(n," 일 때 생성된 수는",gen)
        n = gen     #d(n)의 반복을 위함.
        if gen >10000 or gen == 0 :     #생성된 수가 10000을 넘으면 while문을 break하고 다음 생성자로
            break
        try :
            num.remove(gen)
        except :    #이미 삭제한 숫자라면 다음 d(n)반복으로.
            continue
        
for i in num :
    print(i)

File: 백준 Java/Step7 함수/src/Pro7_2.java
import java.io.*;

public class Pro7_2 {
	public static void main(String[] args) throws IOException {
		BufferedWriter br = new BufferedWriter(new OutputStreamWriter(System.out));
		int[] num = new int[10000];
		int gen = 0;
		for(int i=1; i<10001; i++) {
			num[i-1] = i;
		}
		for(int i=0; i<10000; i++) {
			gen = i + i%10 + i/10%10 + i/100%10 + i/1000%10 + i/10000;
			for(int j=0; j<10000; j++) {//찾아 지우기
				if(num[j] == gen) {
					num[j] = 0;
				}
			}
		}
		for(int i=0; i<10000; i++) {//출력
			if(num[i] != 0) {
				br.write(Integer.toString(num[i]));
				br.newLine();
			}
		}
		br.flush();br.close();
		
//이것도 작동은 잘 되는데 백준에서 런타임에러 나서 BufferdWriter로 바꿈
//		for(int i=0; i<10001; i++) {//출력
//			if(num[i] != 0) {
//				System.out.println(num[i]);
//			}
//		}
		
	}
}
